### DISCUSSION TITLE
Could we support gcc 6.3? Or how to port it to gcc 6.3 and lower version cmake?

### DISCUSSION THREAD

**Discussion by soyoeml** [libp2p/cpp-libp2p#247] [Q&A]
I tried to build this project under gcc 6.3 and cmake 3.7.2. It failed with the following information. Is it possible to let it support gcc 6.3? Please give me some suggestions or steps. Thanks.

```

CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.12 or higher is required.  You are running version 3.7.2


-- Configuring incomplete, errors occurred!
```

**Comment by naivelin** [comment]
The main blocker is soralog now.

CMake Error at cmake/functions.cmake:13 (message):
  [soralog] Requires GNU compiler at least version 8
